src/main/java/backend/Logic.java

```diff
@@ -36,8 +38,9 @@
     private static final String STRING_EMPTY = "";
     private static final char STRING_PERIOD = '.';
 
-    private static final String REGEX_NEITHER_ALPHANUMERIC_NOR_SPACE =
-        "[^ a-zA-Z0-9]+";
+    private static final String REGEX_AUTHOR_NAME_FORMAT = "(^[a-zA-Z0-9][a-zA-Z0-9-]*[a-zA-Z0-9])";
```

Comment:
You're right, if we allow `@@author matric-num----`, which results with the author name being `matric-num`,  `@@author --matric-num----` and `@@author !@%$&*^matric-num` likewise should not fail.